contracts/GroupHealth.sol

```diff
@@ -0,0 +1,297 @@
+     */
+    function setDependencies(
+        address _stakedCelo,
+        address _account,
+        address _specificGroupStrategy,
+        address _defaultStrategy,
+        address _manager
+    ) external onlyOwner {
+        require(_stakedCelo != address(0), "StakedCelo null");
+        require(_account != address(0), "Account null");
+        require(_specificGroupStrategy != address(0), "SpecificGroupStrategy null");
+        require(_defaultStrategy != address(0), "DefaultStrategy null");
+        require(_manager != address(0), "Manager null");
+
+        stakedCelo = IStakedCelo(_stakedCelo);
+        account = IAccount(_account);
+        specificGroupStrategy = ISpecificGroupStrategy(_specificGroupStrategy);
+        defaultStrategy = IDefaultStrategy(_defaultStrategy);
+        manager = IManager(_manager);
+    }
+
+    /**
+     * @notice Returns the storage, major, minor, and patch version of the contract.
+     * @return Storage version of the contract.
+     * @return Major version of the contract.
+     * @return Minor version of the contract.
+     * @return Patch version of the contract.
+     */
+    function getVersionNumber()
+        external
+        pure
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return (1, 1, 0, 0);
+    }
+
+    /**
+     * @notice Checks if a group meets the validator group health requirements.
+     * @param group The group to check for.
+     * @return Whether or not the group is valid.
+     */
+    function isValidGroup(address group) public view returns (bool) {
+        IValidators validators = getValidators();
+
+        // add check if group is !registered
+        if (!validators.isValidatorGroup(group)) {
+            return false;
+        }
+
+        (address[] memory members, , , , , uint256 slashMultiplier, ) = validators
+            .getValidatorGroup(group);
+
+        // check if group has no members
+        if (members.length == 0) {
+            return false;
+        }
+        // check for recent slash
+        if (slashMultiplier < 10**24) {
+            return false;
+        }
+        // check that at least one member is elected.
+        for (uint256 i = 0; i < members.length; i++) {
+            if (isGroupMemberElected(members[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice Returns expected Celo amount voted for by Account contract
+     * vs actual amount voted for by Acccount contract
+     * @param group The group.
+     */
+    function getExpectedAndRealCeloForGroup(address group) public view returns (uint256, uint256) {
+        bool isSpecificGroupStrategy = specificGroupStrategy.isSpecificGroupStrategy(group);
+        bool isActiveGroup = defaultStrategy.groupsContain(group);
+        uint256 realCelo = account.getCeloForGroup(group);
+
+        if (!isSpecificGroupStrategy && !isActiveGroup) {
```

Comment:
Do we need the four cases? Couldn't we just compute the sum, a particular component being 0 if the strategy doesn't apply? Would deduplicate some of the logic.

Reply:
Thanks! Sum is more elegant :-) Updated